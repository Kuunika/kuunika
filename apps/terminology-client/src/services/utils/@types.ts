export interface UIState {
  activePage: string;
}

export interface Category {
  categoryTitle: string;
  id: null | string | number;
  categories: Array<Category>;
}

export interface CategoryData {
  sourceHeadings: Array<string>;
  results: Array<any>;
  formatedData?: Array<any>;
}

export interface Concept {
  id: number | string | null;
  headings: Array<{ title: string; value: string }>;
  descriptions: Array<{ title: string; value: string }>;
  breadcrumb: string;
}
export interface IData {
  categories: Array<Category>;
  categoryData: { [key: string]: CategoryData };
  concept: { [key: string]: Concept };
}

export interface State {
  ui: UIState;
  data: IData;
  errors: any;
  loading: any;
}
