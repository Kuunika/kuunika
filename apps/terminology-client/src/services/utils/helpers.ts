import { Category } from './@types';

export const getSubCategories = (
  categories: Array<Category>,
  targetCategories = [] as Array<string>
) => {
  if (targetCategories.length === 0 || categories.length == 0)
    return categories;

  const targetCategory = categories.find(
    category =>
      category.categoryTitle.toLowerCase() ===
      targetCategories[0].toLowerCase().replace('-', ' ')
  );

  if (targetCategory.categories == null) return categories;

  return getSubCategories(
    targetCategory.categories,
    targetCategories.splice(1)
  );
};
