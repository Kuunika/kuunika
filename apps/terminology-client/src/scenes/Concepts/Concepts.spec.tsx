import React from 'react';
import { render } from '@testing-library/react';

import { ConceptsView } from './index';
import { CategoryData } from '../../services/utils/@types';
import { BrowserRouter } from 'react-router-dom';

describe('Concepts Page', () => {
  it('should render concepts page', () => {
    const data = {
      data: {
        sourceHeadings: ['id', 'name'],
        results: [
          { id: 1, name: 'name' },
          { id: 2, name: 'name2' },
          { id: 3, name: 'name3' }
        ],
        formatedData: [
          { id: 1, name: 'name' },
          { id: 2, name: 'name2' },
          { id: 3, name: 'name3' }
        ]
      } as CategoryData,

      filter: '',
      onChangeSearch: () => {},
      breadCrumb: ['clinical', 'ICD10']
    };
    const wrapper = render(
      <BrowserRouter>
        <ConceptsView {...data} loading={false} />
      </BrowserRouter>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
