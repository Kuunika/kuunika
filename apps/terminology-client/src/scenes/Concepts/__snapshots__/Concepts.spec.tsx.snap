// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Concepts Page should render concepts page 1`] = `
<body>
  <div>
    <div
      class="sc-gqjmRU KqEos"
      data-testid="concepts-table"
    >
      <div
        class="sc-VigVT kebjCN"
      >
        <h2
          class="sc-bdVaJa fWJwsq"
          data-testid="breadcrumb"
        >
          <b>
            Category: 
          </b>
          <span>
            <i
              class="sc-bwzfXH hCGXCx"
            >
              clinical
            </i>
            <i
              class="sc-htpNat diwqqW"
            />
          </span>
          <span>
            <i
              class="sc-bwzfXH hCGXCx"
            >
              ICD10
            </i>
          </span>
        </h2>
        <div
          class="sc-jTzLTM bxVmVK"
        >
          <input
            class="sc-fjdhpX dnLDnU"
            placeholder="Search"
            value=""
          />
          <div
            class="sc-jzJRlG gwYrLW"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-kAzzGY LlbHt"
      >
        <a
          download="Terminologies_Fri Feb 14 2020.csv"
          href="data:text/csv;charset=utf-8,ï»¿\\"id\\",\\"name\\"
\\"1\\",\\"name\\"
\\"2\\",\\"name2\\"
\\"3\\",\\"name3\\""
          target="_self"
        >
          <button
            class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-contained"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Download
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
      <div>
        <table
          class="sc-EHOje bkcjkp"
        >
          <thead>
            <tr
              class="sc-bZQynM epxoqx"
            >
              <th
                class="sc-htoDjs bcgzFl"
              >
                id
              </th>
              <th
                class="sc-htoDjs bcgzFl"
              >
                name
              </th>
            </tr>
            <tr
              class="sc-bZQynM epxoqx"
            />
          </thead>
          <tbody>
            <tr
              class="sc-gzVnrw czazmP"
            >
              <td
                class="sc-dnqmqq kRMmqI"
              >
                1
              </td>
              <td
                class="sc-dnqmqq kRMmqI"
              >
                name
              </td>
            </tr>
            <tr
              class="sc-gzVnrw czazmP"
            >
              <td
                class="sc-dnqmqq kRMmqI"
              >
                2
              </td>
              <td
                class="sc-dnqmqq kRMmqI"
              >
                name2
              </td>
            </tr>
            <tr
              class="sc-gzVnrw czazmP"
            >
              <td
                class="sc-dnqmqq kRMmqI"
              >
                3
              </td>
              <td
                class="sc-dnqmqq kRMmqI"
              >
                name3
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="sc-iwsKbI LsUeq"
        >
          <div>
            1
             to 
            3
             of 
            3
             Records
          </div>
          <div>
            Page 
            1
             of 
            1
            <div
              class="sc-ifAKCX bUHJMM"
            >
              <div
                class="sc-bxivhb kXmlBX"
              >
                1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
